### ISSUE TITLE
Still working?

### ISSUE THREAD

**Issue by Revul** [s4bb4th/HellDivers2Macros#3]
Used this the other day and it worked fine, now when I try it today, nothing happens.  Doesn't seem to register at all in game.

If I try a keyboard tester outside of Hellriders 2 it works fine and triggers the correct buttons.

Did they block this somehow?

**Comment by hbq0449** [comment]
Mine is not working too. looks like nprotect didn't recognize the simulited input or something. Any idea to fix?

**Comment by Siedjeey** [comment]
I have the same issue, yesterday evening it worked fine. Today it doenst work anymore...
Any fix?

**Comment by steakboy** [comment]
Isn't working for me. Keyboard tester says its working, but the game wont register it. Ive tried many different things

**Comment by s4bb4th** [comment]
UPDATE: Was on vaca and just got back. Looks like they patched this method. I will look further into it, but a working solution might take a lot more effort than its worth.

**Comment by Siedjeey** [comment]
> UPDATE: Was on vaca and just got back. Looks like they patched this method. I will look further into it, but a working solution might take a lot more effort than its worth.

Yeah, only way to still use macro's is with Logitech G-Hub currently

**Comment by LordDeimos666** [comment]
Playing on linux with ahk, so nprotect can go to hell

**Comment by Faolon12** [comment]
would be sweet if it can work again, i have problems with the tendons in my hands, and this was helpful and let me play longer,

**Comment by XperimentalOdin** [comment]
Update, Logitech G-Hub got patched out too - devs _really_ don't want stratagems assignable by macro, apparently

**Comment by spyoungtech** [comment]
For a working solution that uses AutoHotkey (via Python), see these examples:

- https://github.com/spyoungtech/helldivers2-macros
- https://github.com/spyoungtech/stratagem-hero-player

Although the code is in Python, it uses AutoHotkey under the hood. This may give you some idea of how to fix this issue. Tested still working as of today.

**Comment by dorkyboi** [comment]
Found somewhere on the internet a different way to send input.
Here's example for Reinforcements, you can copy each key individually to other stratagems.
Do note that there are two instructions per key (`down` action and `up` action), unlike regular `"Send {w}"`

```
Numpad0::
dllcall("keybd_event", int, 162, int, 29, int, 0, int, 0)  ;CTRL DOWN
Sleep, 500
dllcall("keybd_event", int, 87, int, 17, int, 0, int, 0) ;W
Sleep, 50
dllcall("keybd_event", int, 87, int, 17, int, 2, int, 0)
Sleep, 50
dllcall("keybd_event", int, 83, int, 31, int, 0, int, 0) ;S
Sleep, 50
dllcall("keybd_event", int, 83, int, 31, int, 2, int, 0)
Sleep, 50
dllcall("keybd_event", int, 68, int, 32, int, 0, int, 0) ;D
Sleep, 50
dllcall("keybd_event", int, 68, int, 32, int, 2, int, 0)
Sleep, 50
dllcall("keybd_event", int, 65, int, 30, int, 0, int, 0) ;A
Sleep, 50
dllcall("keybd_event", int, 65, int, 30, int, 2, int, 0)
Sleep, 50
dllcall("keybd_event", int, 87, int, 17, int, 0, int, 0) ;W
Sleep, 50
dllcall("keybd_event", int, 87, int, 17, int, 2, int, 0)
Sleep, 50
dllcall("keybd_event", int, 162, int, 29, int, 2, int, 0) ;CTRL UP
return
```

Can confirm that this approach works
